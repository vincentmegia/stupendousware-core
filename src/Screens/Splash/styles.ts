import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
   container: {
      flex: 1,
      marginVertical: 16,
   },
   logoContainer: {
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'flex-end',
   },
   logo: {
      alignSelf: 'center',
      height: 120,
      width: 120,
   },
   textContainer: { flex: 1, justifyContent: 'flex-end' },
   text: { justifyContent: 'flex-end', alignSelf: 'center' },
})

export default styles
