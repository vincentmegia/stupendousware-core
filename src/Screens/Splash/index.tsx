import React from 'react'
import { Image, Text, View } from 'react-native'
import images from '../../assets'
import styles from './style'

const Splash = () => {
  return (
    <View style={styles.container}>
      <View style={styles.logoContainer}>
        <Image source={images.logo} style={styles.logo} />
      </View>
      <View style={styles.textContainer}>
        <Text style={styles.text}>Copyright Stupendousware 2022.</Text>
      </View>
    </View>
  )
}

export default Splash
