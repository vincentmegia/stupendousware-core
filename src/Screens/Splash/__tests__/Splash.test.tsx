import React from 'react'
import { render } from '@testing-library/react-native'
import Splash from '../'

describe('Splash screen test', () => {
   it('matches the snapshot', async () => {
      const { toJSON } = render(
         <Splash onInit={jest.fn} onComplete={jest.fn} />
      )
      expect(toJSON()).toMatchSnapshot()
   })
})
