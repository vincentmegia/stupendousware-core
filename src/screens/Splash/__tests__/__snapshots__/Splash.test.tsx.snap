// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash screen test matches the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }
  >
    <Image
      source=""
      style={
        Object {
          "alignSelf": "center",
          "height": 120,
          "width": 120,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "flex-end",
        }
      }
    >
      Copyright 2022 Stupendousware
    </Text>
  </View>
</View>
`;
